src/containers/BooksForm.js

```diff
@@ -1,19 +1,41 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { CREATE_BOOK } from '../actions';
 
 const categories = ['Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi'];
 
-const BooksForm = () => (
+const BooksForm = () => {
+  const dispatch = useDispatch();
+
+  const [bookName, setBookName] = useState('');
+  const [bookCat, setBookCat] = useState('');
+
+  const handleInput = ({ target: { value } }) => setBookName(value);
+  const handleSelect = ({ target: { value } }) => setBookCat(value);
+
+  const handleSubmit = () => {
+    dispatch(CREATE_BOOK({
+      id: Math.floor(Math.random() * 1000),
```

Comment:
- [ ] Never use a random number to define an id.
  - An id should be ALWAYS unique, the random function can very likely generate the same id if you have a lot of books. If you have two books with the same id that can create a lot of logic errors.
  - Instead create a counter of books created and store it in your store, use that value to set ids and increment that value everytime a book is created.
  - You can also use npm packages created specifically for the creation of unique ids, such as [uuid](https://www.npmjs.com/package/uuid).